packages/federation/src/supergraph.ts

```diff
@@ -72,6 +78,11 @@ function getTypeFieldMapFromSupergraphAST(supergraphAST: DocumentNode) {
   return typeFieldASTMap;
 }
 
+const rootTypeMap = new Map<string, OperationTypeNode>([
+  ['Query', 'query' as OperationTypeNode],
+  ['Mutation', 'mutation' as OperationTypeNode],
+  ['Subscription', 'subscription' as OperationTypeNode],
+]);
```

Comment:
Much better, if legacy GQL-JS agrees.
```suggestion
const rootTypeMap = new Map([
  ['Query', OperationTypeNode.QUERY],
  ['Mutation', OperationTypeNode.MUTATION],
  ['Subscription', OperationTypeNode.SUBSCRIPTION],
]);
```

Reply:
Unfortunately not :/ because they are a union of strings

Reply:
Alright.